module/mempool/queue/heroQueue.go

```diff
@@ -0,0 +1,71 @@
+package queue
+
+import (
+	"sync"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+	herocache "github.com/onflow/flow-go/module/mempool/herocache/backdata"
+	"github.com/onflow/flow-go/module/mempool/herocache/backdata/heropool"
+)
+
+// HeroQueue implements a HeroCache-based in-memory queue.
+type HeroQueue struct {
+	mu        sync.RWMutex
+	cache     *herocache.Cache
+	sizeLimit uint
+}
+
+func NewHeroQueue(sizeLimit uint32, logger zerolog.Logger, collector module.HeroCacheMetrics) *HeroQueue {
+	return &HeroQueue{
+		cache: herocache.NewCache(
+			sizeLimit,
+			herocache.DefaultOversizeFactor,
+			heropool.NoEjection,
+			logger.With().Str("mempool", "hero-queue").Logger(),
+			collector),
+		sizeLimit: uint(sizeLimit),
+	}
+}
+
+// Push stores the entity into the queue.
+// Boolean returned variable determines whether push was successful, i.e.,
+// push may be dropped if queue is full or already exists.
```

Comment:
Now we reuse the same heroQueue module to back the providers in collection and execution nodes. And they also have the same config. 
```
	// DefaultRequestProviderWorkers is the default number of workers used to process entity requests.
	DefaultRequestProviderWorkers = uint(5)

	DefaultEntityRequestCacheSize = 1000
```

However, the number of requests they are serving might be very different. I'm OK to use the same config, but it's better we know whether these config is fine turned. 

For that, we probably need some metrics to know:
- the speed of adding jobs to the queue, meaning the message request is received and waiting to be processed.
- the speed of removing jobs to the queue, meaning the messages have been processed
- the number of jobs remaining in the queue. 
- whether there is message dropped from the queue when it's full

Do we have visibility over the above metrics?

Reply:
`HeroCache` provides the following metrics: 

> the speed of adding jobs to the queue, meaning the message request is received and waiting to be processed.

[KeyPutAttempt](https://github.com/onflow/flow-go/blob/ba75711b2133ad556fdf571b92bda4e28f33535a/module/metrics/herocache.go#L113)

> the speed of removing jobs to the queue, meaning the messages have been processed

[KeyRemoved](https://github.com/onflow/flow-go/blob/ba75711b2133ad556fdf571b92bda4e28f33535a/module/metrics/herocache.go#L141)

> the number of jobs remaining in the queue.

[Size](https://github.com/onflow/flow-go/blob/ba75711b2133ad556fdf571b92bda4e28f33535a/module/metrics/herocache.go#L92)

> whether there is message dropped from the queue when it's full.

[KeyPutDrop](https://github.com/onflow/flow-go/blob/ba75711b2133ad556fdf571b92bda4e28f33535a/module/metrics/herocache.go#L120)